src/com/geekhub/hw5/translator/Translator.java

```diff
@@ -0,0 +1,80 @@
+package com.geekhub.hw5.translator;
+
+import com.geekhub.hw5.translator.source.URLSourceProvider;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Provides utilities for translating texts to russian language.<br/>
+ * Uses Yandex Translate API, more information at <a href="http://api.yandex.ru/translate/">http://api.yandex.ru/translate/</a><br/>
+ * Depends on {@link URLSourceProvider} for accessing Yandex Translator API service
+ */
+public class Translator {
+    private URLSourceProvider urlSourceProvider;
+    /**
+     * Yandex Translate API key could be obtained at <a href="http://api.yandex.ru/key/form.xml?service=trnsl">http://api.yandex.ru/key/form.xml?service=trnsl</a>
+     * to do that you have to be authorized.
+     */
+    private static final String YANDEX_API_KEY = "trnsl.1.1.20151114T120610Z.e08b88aa89ea5295.90f276ad4cd8e526e2989a2182a0ec2b3c742f89";
+    private static final String TRANSLATION_DIRECTION = "ru";
+
+    public Translator(URLSourceProvider urlSourceProvider) {
+        this.urlSourceProvider = urlSourceProvider;
+    }
+
+    /**
+     * Translates text to russian language
+     * @param original text to translate
+     * @return translated text
+     * @throws IOException
+     */
+    public String translate(String original) throws IOException {
+        String translated = urlSourceProvider.load(prepareURL(original));
+        return parseContent(translated);
+//        return translated;
+    }
+
+    /**
+     * Prepares URL to invoke Yandex Translate API service for specified text
+     * @param text to translate
+     * @return url for translation specified text
+     */
+    private String prepareURL(String text) {
+        return "https://translate.yandex.net/api/v1.5/tr/translate?key=" + YANDEX_API_KEY + "&text=" + encodeText(text) + "&lang=" + TRANSLATION_DIRECTION;
+    }
+
+    /**
+     * Parses content returned by Yandex Translate API service. Removes all tags and system texts. Keeps only translated text.
+     * @param content that was received from Yandex Translate API by invoking prepared URL
+     * @return translated text
+     */
+    private String parseContent(String content) {
+        String regexp = "<text>(.+)?</text>";
```

Comment:
Some problems with this regexp.. In some cases it doesnt recognize "</text>" part of file. May be it is better to use some XML parser java classes?